### PR TITLE
build: package only translations explicitly marked as supported

### LINKED ISSUE #1438: Add list of supported languages; disable unsupported languages
SecureDrop Client translation is currently unofficially supported via Weblate (it has not been announced/publicized yet); as a consequence, some volunteers have started translating the client into other languages. We do even have some 100% completions already:

https://weblate.securedrop.org/projects/securedrop/securedrop-client/

The way it works right now, however, you can run the app in a partially translated language. For example, as of this writing, Catalan is at 69%, and you'll see a partially translated UI if you launch it with `LANG=ca`.

Ideally, we should only enable languages in the app once they've reached full completion, and only disable them once they fall below a certain threshold (e.g., 80%). To do so, we need to be able to specify a list of languages that are supported, vs. ones that are currently unsupported. This should _not_ prevent translation in Weblate, it should only prevent the language from being made available in the app.

If we're unable to initially support certain languages for technical reasons (e.g., RTL languages, see #1406), we can also use that capability to delay enabling those languages until technical foundations are in place.

### PR DESCRIPTION
**Pull request by cfm**
# Description

Closes #1438 by adding in `MANIFEST.in` (rather than `setup.py` as originally proposed):
1. a `prune` directive to exclude translation assets present in the tree by default; and
2. instructions for adding `graft` directives for explicitly including translation assets only for languages considered supported.

Also:

3. removes `make supported-languages`, obsoleted by this approach per <https://github.com/freedomofpress/securedrop-client/issues/1438#issuecomment-1067437258>;
4. removes the Weblate CLI (`wlc`), which is no longer needed; and
5. updates development dependencies along the way.
    * ebb5336 fixes an unrelated typing error caught by (4)'s update of mypy 0.940 → 0.960.  Let me know if a stack of pull requests is preferable for separate review of (3)‒(4) and _then_ these (1)‒(2).

# Test Plan

- [ ] By default, no translations are packaged:
    ```sh-session
    (.venv) user@sd-dev:~/securedrop-client$ python setup.py sdist 2> /dev/null | grep "locale"
    (.venv) user@sd-dev:~/securedrop-client$
    ```
- Pick a language from `securedrop_client/locale`, e.g.:
    ```sh-session
    (.venv) user@sd-dev:~/securedrop-client$ export LANG=de
    ```
- [ ] An unpackaged translation is still loadable:
    ```sh-session
    (.venv) user@sd-dev:~/securedrop-client$ python -m securedrop_client  # appears in $LANG
    ```
- [ ] A language marked for inclusion is packaged:
    ```sh-session
    (.venv) user@sd-dev:~/securedrop-client$ echo "graft securedrop_client/locale/$LANG" >> MANIFEST.in
    (.venv) user@sd-dev:~/securedrop-client$ python setup.py sdist 2> /dev/null | grep "locale"
    creating securedrop-client-0.7.0/securedrop_client/locale
    creating securedrop-client-0.7.0/securedrop_client/locale/de
    creating securedrop-client-0.7.0/securedrop_client/locale/de/LC_MESSAGES
    copying securedrop_client/locale/de/LC_MESSAGES/messages.mo -> securedrop-client-0.7.0/securedrop_client/locale/de/LC_MESSAGES
    copying securedrop_client/locale/de/LC_MESSAGES/messages.po -> securedrop-client-0.7.0/securedrop_client/locale/de/LC_MESSAGES
    ```

# Checklist

If these changes modify code paths involving cryptography, the opening of files in VMs or network (via the RPC service) traffic, Qubes testing in the staging environment is required. For fine tuning of the graphical user interface, testing in any environment in Qubes is required. Please check as applicable:

 - I have tested these changes in the appropriate Qubes environment
 - I do not have an appropriate Qubes OS workstation set up (the reviewer will need to test these changes)
 - [x] These changes should not need testing in Qubes

If these changes add or remove files other than client code, the AppArmor profile may need to be updated. Please check as applicable:

 - I have updated the [AppArmor profile](https://github.com/freedomofpress/securedrop-client/blob/HEAD/files/usr.bin.securedrop-client)
 - [x] No update to the AppArmor profile is required for these changes

These assets are covered by:
https://github.com/freedomofpress/securedrop-client/blob/1c5482b7b4254d09818c7f829dff6f30c5955792/files/usr.bin.securedrop-client#L21

 - I don't know and would appreciate guidance

If these changes modify the database schema, you should include a database migration. Please check as applicable:

 - I have written a migration and upgraded a test database based on `main` and confirmed that the migration applies cleanly
 - I have written a migration but have not upgraded a test database based on `main` and would like the reviewer to do so
 - I need help writing a database migration
 - [x] No database schema changes are needed

### CODE REVIEW

**Comment by cfm** [comment]
Once CI has been placated on `main`, I'll rebase this and see if anything's still unhappy here.

**Comment by gonzalo-bulnes** [comment]
You should be good to go whenever you want now @cfm -[CI is fixed | latest nightly build](https://app.circleci.com/pipelines/github/freedomofpress/securedrop-client/2089/workflows/9aa951d7-d957-4d53-b628-ab19c675e556) 🙂

**Comment by cfm** [comment]
Ready for review with CI (finally) passing.  Along the way, ebb5336 fixes an unrelated typing error caught by new mypy 0.940 → 0.960.

This is an unfortunate side effect of _not_ pinning development dependencies.  Removing a dependency X from a `requirements.in` still entails regenerating the `requirements.txt` for all dependencies X'.  And new versions of mypy often ~break~ _reveal preexisting bugs_ in things!

Let me know if a stack of pull requests is preferable for separate review.

File: MANIFEST.in [hunk 1]
```diff
@@ -16,3 +16,16 @@ recursive-include securedrop_client *
...
+
+# Translations may appear in the tree before we consider them supported, so
+# don't include any by default.
+prune securedrop_client/locale
+
+# Explicitly include only supported translations, one line for each $LANG, like:
+#
+#     graft securedrop_client/locale/$LANG
+#
+# Please keep this list alphabetized.
```
Comment (creviera) [inline MANIFEST.in:31]: <3

File: securedrop_client/queue.py [hunk 2]
```diff
@@ -92,7 +92,8 @@ def _check_for_duplicate_jobs(self, job: QueueJob) -> bool:
         stored on self.current_job. We check that the job to be added is not among them.
         """
         in_progress_jobs = [in_progress_job for priority, in_progress_job in self.queue.queue]
-        in_progress_jobs.append(self.current_job)
+        if self.current_job is not None:
+            in_progress_jobs.append(self.current_job)
```
Comment (creviera) [inline securedrop_client/queue.py:96]: I see this was caught by mypy update to 0.960

**Review by creviera — APPROVED** [review]
lgtm - test plan checks out

**Pull request merged**